### ISSUE TITLE
Importer: MetaData with empty values should not be stored

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/OSPSuite.Core#1429]
[TestData.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/7828854/TestData.xlsx)

In MoBi:

- Import the sheet "Mitrakou_1990" for import and use the following mapping:

![image](https://user-images.githubusercontent.com/25903040/148546613-f52d5f38-a96d-4eae-a737-7293c0fb8a33.png)

In the original file, the columns "Study Id" and "Patient Id" do not contain any values:

![image](https://user-images.githubusercontent.com/25903040/148546710-5bd2493d-15b8-4126-aee0-8dc96ccd9da8.png)

In the imported data sets, there are errors next to those meta data:

![image](https://user-images.githubusercontent.com/25903040/148546776-94ae38ae-f3fb-444a-80f3-a91e4c2e241b.png)

The  values should be just empty, but not showing an error (e.g. an empty string).

**Comment by msevestre** [comment]
I find this proposal a bit strange.
If we want to allow mapping to empty column, then we should NOT create the meta data instead of allowing meta data to have no values.

**Comment by msevestre** [comment]
But I am questioning the fact that the importer allows mapping empty column....I would like to understand the use case again

**Comment by PavelBal** [comment]
> If we want to allow mapping to empty column, then we should NOT create the meta data instead of allowing meta data to have no values.

I am ok with it.

> But I am questioning the fact that the importer allows mapping empty column....I would like to understand the use case again

The greatest thing about the new importer is the re-usability. We can define one excel sheet structure, create a configuration for it, save it to xml, and re-use from here on.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File href="file:///C:/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List href="file:///C:/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



Study Id | Subject Id | Organ | Compartment | Species | Gender | Dose | Molecule | Molecular Weight | Time | Time unit | Measurement | Measurement unit | Error | LLOQ | Route | Group Id
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
ESQ-ID-001 | 001-002 | Venous Blood | PLASMA | Human | MALE | Dapa 5 mg QD | Glucose | 180 | 0.00 | h | 4.8 | mmol/l | 0.3 | 0.01 | po |  



</body>

</html>


From now on, I can use this sheet for every possible data set, and when I import it, I only need to select the excel file and to load the configuration - it works!

Now if I for example do not have the information about the Subject Id, I just leave this row empty. Otherwise I would have to define a new sheet structure (without the column "Subject Id"), and use it with another configuration file. And now imagine I have the information about Subject Id (just an example - could be anything else) later - I would have add a new column to the existing sheets and use another configuration?

In short - it is "one sheet structure with one configuration for everything" versus "separate sheet structures and configurations for every possible combination of available information".

**Comment by georgeDaskalakis** [comment]
The error is not reproducible in Core Starter with the current develop version. Maybe it should be checked once again and maybe then close it.

**Comment by PavelBal** [comment]
never say never...

**Comment by msevestre** [comment]
@georgeDaskalakis Not fixed  using latest code in develop (11.0.60)
![image](https://user-images.githubusercontent.com/1041237/151422582-fe472a77-a3af-451b-bb50-6db1bd694cde.png)
